### PR TITLE
Export versioning support

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose
Add support for versioned spec files in spec2def and add vista exports to kernel32.spec

### CODE REVIEW

File: sdk/tools/spec2def/spec2def.c [hunk 1]
```diff
@@ -49,6 +49,7 @@ char *pszSourceFileName = NULL;
 char *pszDllName = NULL;
 char *gpszUnderscore = "";
 int gbDebug;
+unsigned guOsVersion = 0x502;
```
Comment (HBelusca) [inline sdk/tools/spec2def/spec2def.c]: Don't we have the _WIN32_WINNT macros available also for the host-tools? (you would specify an explicit _WIN32_WINNT_WS03 here for example).

Comment (tkreuzer) [reply sdk/tools/spec2def/spec2def.c]: We don't

**Comment by yagoulas** [comment]
What are you up to Timo?

**Comment by tkreuzer** [comment]
The idea is to allow building sets of "wrapper DLLs", one for each OS version. They all forward their exports to a base version (like "roskernelbase.dll") except for those that need special handling. All this can be done without creating half a dozen of copies of the spec file.

**Comment by learn-more** [comment]
This will make it more complicated to make apisets?
Because we will need apisets for each version starting from 7?

Or would the idea be, that the apisets forward to the first version that has a specific function?

**Comment by tkreuzer** [comment]
@learn-more Why would it make apisets more complicated? If we implement all functions for all versions in a core-dll, then all apiset dlls can forward to that directly.

**Comment by learn-more** [comment]
it means we also have to propagate min/max versions to all spec files.

**Comment by tkreuzer** [comment]
@learn-more Only when adding new exports, otherwise why would you need that? The default is win2k3, so without adding any Vista+ exports everything is exactly as before.

**Comment by encodedpr** [comment]
any blockers remain on this PR?

**Pull request merged**